### PR TITLE
NO-ISSUE: Bump go-openapi/runtime to 0.23.2

### PR DESCRIPTION
**Pull request by danielerez**
Updating runtime to 0.23.2 cuases RegsiterHost API to fail [1].
The root cause is a change [2] in runtime package, in which url.Parse is used on the basePath.
So since the agent is adding double slashes in the client url, e.g. <inventory>//api/assisted-install [3], url.Parse omits the '//api' prefix.

I.e.
basePathURL, _ := url.Parse("//api/assisted-install")
basePathURL.Path --> /assisted-install  (instead of /api/assisted-install).

Hence, removed the redundant '/' in createUrl func template to avoid API request failure.

[1] "Error registering host: json: cannot unmarshal number into Go struct field Error.code of type string"
[2] https://github.com/go-openapi/runtime/commit/872b746e6eaeb1de7dedc59f746e3507d2a053c5#diff-d1c4178bc92354ebb52bb3b7c282de3de06d31949d9fbd7a77e74e52dbcb4842R285
[3] https://github.com/openshift/assisted-installer-agent/blob/master/src/session/inventory_session.go#L32

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,6 +25,7 @@ require (
 	github.com/stretchr/testify v1.7.0
 	github.com/thoas/go-funk v0.9.1
 	github.com/vishvananda/netlink v1.1.0
+	go.mongodb.org/mongo-driver v1.8.4 // indirect
```
[the remark this answers is not in the document]
Comment (danielerez) [reply go.mod]: It's used indirectly by the runtime package: https://github.com/go-openapi/runtime/blob/753b551e6b4a36e461ac877874fd0a79d0ffcf53/go.mod#L18

File: src/session/inventory_session.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ import (
 )
 
 func createUrl(inventoryUrl string) string {
-	return fmt.Sprintf("%s/%s", inventoryUrl, client.DefaultBasePath)
+	return fmt.Sprintf("%s%s", inventoryUrl, client.DefaultBasePath)
```
Comment (avishayt) [inline src/session/inventory_session.go:32]: path.Join is safer than fmt.Sprintf

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (filanov) [reply go.mod:28]: strange that it adds it here and not just in go.sum

File: src/session/inventory_session.go (hunk 2 above)
Comment (tsorya) [reply src/session/inventory_session.go:32]: yep, especially because inventoryUrl is configurable and agent got it from agent

Comment (danielerez) [reply src/session/inventory_session.go]: changed, just needed some error handling.

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielerez) [reply go.mod]: yeah it was added when updating the package. i'll try to manually remove it from go.mod, it should stay on go.sum.

**Pull request merged**